src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorConfigurationService.cs

```diff
@@ -84,25 +74,48 @@ public async override Task<RazorLSPOptions> GetLatestOptionsAsync(CancellationTo
             }
         }
 
-        private RazorLSPOptions BuildOptions(IConfiguration config)
+        private static RazorLSPOptions BuildOptions(JObject[] result)
         {
             var instance = RazorLSPOptions.Default;
 
-            Enum.TryParse(config["razor:trace"], out Trace trace);
+            var razor = result[0];
+            var html = result[1];
+            var editor = result[2];
+
+            var trace = instance.Trace;
+            if (razor.TryGetValue("trace", out var parsedTrace))
+            {
+                trace = parsedTrace.ToObject<Trace>();
+            }
 
             var enableFormatting = instance.EnableFormatting;
-            if (bool.TryParse(config["razor:format:enable"], out var parsedEnableFormatting))
+            if (razor.TryGetValue("format", out var parsedFormat))
             {
-                enableFormatting = parsedEnableFormatting;
+                if (((JObject)parsedFormat).TryGetValue("enable", out var parsedEnableFormatting))
```

Comment:
This JObject cast might crash if someone hasn't fully filled out their `format` section in VSCode. You'll want to conditionally cast to JOBject